src/ristretto/pedersen/extended_commitment_factory.rs

```diff
@@ -0,0 +1,451 @@
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use bulletproofs_plus::{generators::pedersen_gens::ExtensionDegree, PedersenGens};
+use curve25519_dalek::{ristretto::RistrettoPoint, scalar::Scalar, traits::Identity};
+
+use crate::{
+    commitment::{ExtendedHomomorphicCommitmentFactory, HomomorphicCommitment},
+    errors::RangeProofError,
+    ristretto::{
+        constants::{RISTRETTO_NUMS_POINTS, RISTRETTO_NUMS_POINTS_COMPRESSED},
+        pedersen::{
+            PedersenCommitment,
+            RISTRETTO_PEDERSEN_G,
+            RISTRETTO_PEDERSEN_G_COMPRESSED,
+            RISTRETTO_PEDERSEN_H,
+            RISTRETTO_PEDERSEN_H_COMPRESSED,
+        },
+        RistrettoPublicKey,
+        RistrettoSecretKey,
+    },
+};
+
+/// Generates extended Pederson commitments `sum(k_i.G_i) + v.H` using the provided base
+/// [RistrettoPoints](curve25519_dalek::ristretto::RistrettoPoints).
+/// Notes:
+///  - Homomorphism with public key only holds for extended commitments with `ExtensionDegree::Zero`
+#[derive(Debug, PartialEq, Clone)]
+pub struct ExtendedPedersenCommitmentFactory(pub(crate) PedersenGens<RistrettoPoint>);
+
+impl ExtendedPedersenCommitmentFactory {
+    /// Create a new Extended Pedersen Ristretto Commitment factory for the required extension degree using
+    /// pre-calculated compressed constants - we only hold references to the static generator points.
+    pub fn new_with_extension_degree(extension_degree: ExtensionDegree) -> Result<Self, RangeProofError> {
+        if extension_degree as usize > RISTRETTO_NUMS_POINTS.len() ||
+            extension_degree as usize > RISTRETTO_NUMS_POINTS_COMPRESSED.len()
+        {
+            return Err(RangeProofError::ExtensionDegree(
+                "Not enough Ristretto NUMS points to construct the extended commitment factory".to_string(),
+            ));
+        }
```

Comment:
I guess we can generalise this in future. But fine for now.

Reply:
Logged issue [here](https://github.com/tari-project/tari-crypto/issues/101).